### PR TITLE
Now compiles with stackage nightly 2016-06-15 (ghc-8.0.1)

### PR DESCRIPTION
**Pull request by ocramz**
Updated a call to VarI (recent changes in Template Haskell)

### CODE REVIEW

**Comment by ocramz** [comment]
Travis build failure is due to `cairo` in the gsl-ode example

**Comment by bitonic** [comment]
Thanks! However:

1. We should not drop support for previous compilers, please use CPP to decide what to do in that function based on the GHC version
2. If `cairo` is not in the nightly, we should disable that example in Travis for the nightly for now.

File: src/Language/C/Inline/FunPtr.hs [hunk 1]
```diff
@@ -12,6 +13,10 @@ import           Foreign.Ptr (FunPtr)
 import qualified Language.Haskell.TH as TH
 import qualified Language.Haskell.TH.Syntax as TH
 
+#ifndef MIN_VERSION_template_haskell
```
Comment (bitonic) [inline src/Language/C/Inline/FunPtr.hs]: Why is this needed?

File: .travis.yml [hunk 2]
```diff
@@ -20,6 +20,7 @@ env:
 - ARGS="--resolver lts-3 --flag inline-c:gsl-example"
 - ARGS="--resolver lts --flag inline-c:gsl-example"
 - ARGS="--resolver nightly --flag inline-c:gsl-example"
+- ARGS="--resolver nightly"
```
Comment (bitonic) [inline .travis.yml]: Maybe we should remove the previous line, with a comment on why the gsl example doesn't work with nightly?

**Comment by ocramz** [comment]
Still learning to develop Haskell in-the-large, thanks for the pointers!

File: stack.yaml [hunk 3]
```diff
@@ -2,4 +2,5 @@ packages:
 - '.'
 - sample-cabal-project/
 extra-deps: []
-resolver: lts-4.1
+# resolver: lts-4.1
+resolver: nightly-2016-06-15
```
Comment (bitonic) [inline stack.yaml]: Since the cairo example does not work with nightly, I wouldn't have it as the default one. Feel free to bump it to lts 6 though.

File: .travis.yml [hunk 4]
```diff
@@ -15,11 +15,10 @@ addons:
 # The different configurations we want to test. You could also do things like
 # change flags or use --stack-yaml to point to a different file.
 env:
-- ARGS="--flag inline-c:gsl-example"
```
Comment (bitonic) [inline .travis.yml]: Once you've restored lts in the `stack.yaml`, this line should be reinstated.

File: .travis.yml [hunk 5]
```diff
@@ -15,11 +15,10 @@ addons:
...
 - ARGS="--resolver lts-2"
 - ARGS="--resolver lts-3 --flag inline-c:gsl-example"
 - ARGS="--resolver lts --flag inline-c:gsl-example"
-- ARGS="--resolver nightly --flag inline-c:gsl-example"
+- ARGS="--resolver nightly"
```
Comment (bitonic) [inline .travis.yml]: Please add a comment saying why the GSL example is not enabled for nightly.

**Comment by ocramz** [comment]
All green now : )

**Comment by bitonic** [comment]
Thanks a lot!

**Comment by ocramz** [comment]
@bitonic My pleasure; could you upload it to hackage? I'd like to use it as soon as possible : ) Thanks

**Comment by bitonic** [comment]
@ocramz I've uploaded a new version.

**Comment by ocramz** [comment]
Thanks! I've sent a PR for the build-constraints, hopefully it will make it into tomorrow's nightly ^_^
